### ISSUE TITLE
[IceHive] - HiveCleaning::BuildLookUpTables segfaults with IceACT in geometry (Trac #2134)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2126]
IceHive segfaults when IceACT is included in the geometry. It assumes the strings are >0 when calculating the hash function: http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/IceHive/trunk/public/IceHive/OMKeyHash.h#L64

see:
https://icecube-spno.slack.com/archives/C02KQL9KN/p1518116966000686

Migrated from https://code.icecube.wisc.edu/ticket/2134
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:13", 
    "description": "IceHive segfaults when IceACT is included in the geometry. It assumes the strings are >0 when calculating the hash function: http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/IceHive/trunk/public/IceHive/OMKeyHash.h#L64\n\nsee:\nhttps://icecube-spno.slack.com/archives/C02KQL9KN/p1518116966000686\n\n", 
    "reporter": "kjmeagher", 
    "cc": "mzoll", 
    "resolution": "fixed", 
    "_ts": "1550067313248429", 
    "component": "combo reconstruction", 
    "summary": "[IceHive] - HiveCleaning::BuildLookUpTables segfaults with IceACT in geometry", 
    "priority": "critical", 
    "keywords": "segfault icehive hivesplitter", 
    "time": "2018-02-08T19:42:57", 
    "milestone": "", 
    "owner": "kkrings", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180208T21:02:32: 
* nega changed summary from "[IceHive]" to "[IceHive] - HiveCleaning::BuildLookUpTables segfaults with IceACT in geometry"
* nega commented:

> On trunk, [source:IceCube/projects/HiveSplitter/trunk HiveSplitter] handles a similar situation by pre-filtering in [source:IceCube/projects/HiveSplitter/trunk/private/HiveSplitter/HiveSplitter.cxx@147228:517-520#L511 private/HiveSplitter/HiveSplitter.cxx:517]. Myabe filtering by OMType would be more elegant?
> 


* nega changed status from "new" to "assigned"
* nega changed keywords from "" to "segfault icehive hivesplitter"
* nega changed cc from "" to "mzoll"
* nega changed owner from "" to "kkrings"

**Comment by icecube-trac** [comment]
Trac update at 20180607T16:43:52: olivas changed milestone from "Vernal Equinox 2018" to "Summer Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20180702T16:20:09: 
* kkrings commented:

> In changeset 163823:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="163823"
> Ignore OMKeys that are not hashable when building the look-up table; closes #2126.
> ```


* kkrings changed status from "assigned" to "closed"
* kkrings changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:13: 
* olivas commented:

> Milestone Summer Solstice 2018 deleted


* olivas changed milestone from "Summer Solstice 2018" to ""
